test/retained_topic_map.hpp

```diff
@@ -197,6 +198,10 @@ class retained_topic_map {
     bool erase_topic(MQTT_NS::string_view topic) {
         auto path = find_topic(topic);
 
+        // Reset the value
+        if(!path.empty())
+            path.back()->second.value = MQTT_NS::optional<Value>();
```

Comment:
```suggestion
            path.back()->second.value = MQTT_NS::nullopt;
```

It's not mandatory but more comprehensible.

See https://github.com/redboltz/mqtt_cpp/blob/master/include/mqtt/optional.hpp#L35

Reply:
Yes, it is annoying you have to name the type to reset the value for MQTT_NS::optional<Value>()